pkg/apis/wmc/v1alpha1/windowsmachineconfig_types.go

```diff
@@ -11,28 +11,24 @@ import (
 type WindowsMachineConfigSpec struct {
 	// Replicas represent how many Windows nodes to be added to the
 	// OpenShift cluster
-	Replicas int `json:"replicas,omitempty"`
+	Replicas int `json:"replicas"`
 	// InstanceType represents the flavor of instance to be used while
 	// creating the virtual machines. Please note that this is common
 	// across all the Windows nodes in the cluster
-	InstanceType string `json:"instanceType,omitempty"`
-	// AWS holds AWS specific cloud provider information.
-	AWS *AWS `json:"aws"`
+	InstanceType string `json:"instanceType"`
+	// AWS holds AWS specific cloud provider information
+	// +optional
+	AWS *AWS `json:"aws,omitempty"`
 	// Azure holds Azure specific cloud provider information
-	Azure *Azure `json:"azure"`
-	// CloudProviderCredentials is the name of the secret which contains the
-	// credentials of the cloud provider
-	CloudProviderCredentials string `json:"cloudProviderCredentials,omitempty"`
+	// +optional
+	Azure *Azure `json:"azure,omitempty"`
 }
 
 // AWS holds the information related to AWS cloud provider
 type AWS struct {
 	// SSHKeyPair is the sshKeyPair associated with cloudprovider. AWS
 	// asks a keypair to be present for encrypting the Windows VM password
 	SSHKeyPair string `json:"sshKeyPair"`
-	// SSHPrivateKey is the name of the secret which contains ssh key
```

Comment:
Since we assume the `SSHPrivateKey` exists in `/etc/private-key/private-key.pem`

Reply:
How does this get added to the operator image?

Reply:
I think, we decided that the user has to manually mount the secret in the deployment.yaml's template spec before the operator gets deployed. I am creating a card on how to use it.

Reply:
How is that going to work when we are using OLM? Please call out in your card that this issue needs to be solved too.

Reply:
> How is that going to work when we are using OLM?

I think, we decided to have a validation which ensures operator fails to come up when the secret is not mounted at that path. It'll be similar to how operator fails to come up when the required CRD is not available. We should still be able to deploy via OLM but we'll tell to update the deployment.yaml with necessary secret. 

>  Please call out in your card that this issue needs to be solved too.

Sure. will do that.

Reply:
https://issues.redhat.com/browse/WINC-316

Reply:
The only point of waiting for the Jira story to be created before addressing other comments in the PR was if it was to be mentioned somewhere in the code. And I don't see it being mentioned anywhere.

Reply:
@ravisantoshgudimetla says "Actually it gets mentioned in the code here: https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/controller/wellknownlocations/wellknownlocations.go#L9"

The please add a link there in this PR.

Reply:
Done.